tests/end-to-end/backup-restore-test/utils/waiter/waiter.go

```diff
@@ -0,0 +1,26 @@
+package waiter
+
+import (
+	"fmt"
+	"log"
+	"time"
+)
+
+func WaitAtMost(fn func() (bool, error), duration time.Duration) error {
+	timeout := time.After(duration)
+	tick := time.Tick(500 * time.Millisecond)
+
+	for {
+		ok, err := fn()
+		select {
+		case <-timeout:
+			return fmt.Errorf("waiting for resource failed in given timeout %f second(s)", duration.Seconds())
+		case <-tick:
+			if err != nil {
+				log.Println(err)
```

Comment:
in case this error is ok to happen, you could store it in a variable and return it in the timeout case instead of spamming the log with the error message.

so the error would be automatically printed by go test in case the test fails, but if the test succeeds nobody would ever have to dig through these messages